src/main/kotlin/uk/gov/justice/digital/hmpps/prisonercontactregistry/controller/PrisonerContactController.kt

```diff
@@ -73,11 +78,72 @@ class PrisonerContactController(
-    return orderByLastNameAndThenFirstName(contactService.getContactList(prisonerId, contactType, personId, withAddress))
+    return contactService.getContactList(prisonerId, contactType, personId, withAddress)
+      .sortedWith(getDefaultSortOrder())
+  }
+
+  @PreAuthorize("hasRole('PRISONER_CONTACT_REGISTRY')")
+  @GetMapping(PRISON_GET_APRROVED_SOCIAL_CONTACTS_CONTROLLER_PATH)
+  @Operation(
+    summary = "Get Prisoners approved Social Contacts",
+    description = "Returns details of a prisoner's social contacts that have been approved.",
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Prisoner Social Contacts Information Returned",
+      ),
+      ApiResponse(
+        responseCode = "400",
+        description = "Incorrect request to Get Prisoner Contacts for Prisoner Identifier",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized to access this endpoint",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Incorrect permissions retrieve a Prisoner Contacts",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "404",
+        description = "Prisoner not found",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+    ],
+  )
+  fun getPrisonersApprovedSocialContacts(
+    @Schema(description = "Prisoner Identifier (NOMIS Offender No)", example = "A1234AA", required = true)
+    @PathVariable
+    prisonerId: String,
+    @RequestParam(value = "id", required = false)
+    @Parameter(description = "Query by Person Identifier (NOMIS Person ID)", example = "9147510")
+    personId: Long?,
+    @RequestParam(value = "hasDateOfBirth", required = false)
+    @Parameter(description = "Defaults to null. By default (or when false), returns all contacts with or without a DOB. If true, returns only contacts with a DOB.", example = "false")
+    hasDateOfBirth: Boolean? = null,
+    @RequestParam(value = "notBannedBeforeDate", required = false)
+    @Parameter(description = "Get only visitors that have a ban date that expires before this date. Gets all visitors irrespective of BANs if not passed.", example = "2024-12-31", required = false)
+    notBannedBeforeDate: LocalDate? = null,
+    @RequestParam(value = "withAddress", required = false)
+    @Parameter(description = "by default returns addresses for all contacts, set to false if contact addresses not needed.", example = "false")
+    withAddress: Boolean? = true,
```

Comment:
do we need with address,.. do we actually use the address?

Reply:
the address is used on staff UI when showing visitors.